promql/engine.go

```diff
@@ -1064,6 +1068,10 @@ type evaluator struct {
 	samplesStats             *stats.QuerySamples
 	noStepSubqueryIntervalFn func(rangeMillis int64) int64
 	enableDelayedNameRemoval bool
+	querier                  storage.Querier
```

Comment:
If the arguments of info are step invariant like `info(foo @ 10, {data=~".+"} @ 10)` then info is stepinvariant too and will be evaluated with an evalhelper without querier because we are not passing it down in that code path leading to a panic. I cannot comment at the specific line since it was not changed and github doesnt allow me but its line 2040.

Reply:
Thank you!

Reply:
I think same goes for subqueries, ill write a couple tests with subqueries to confirm later.

Reply:
@MichaHoffmann do you know how to obtain a querier for line 2040?

Reply:
Fixed for step invariation + sub-query.